lib/fluent/plugin/out_forward.rb

```diff
@@ -672,24 +838,42 @@ def send_data_actual(sock, tag, chunk)
       end
 
       def send_data(tag, chunk)
-        sock = @sender.create_transfer_socket(resolved_host, port, @hostname)
+        sock = connect
+
         begin
           send_data_actual(sock, tag, chunk)
         rescue
-          sock.close rescue nil
+          if @keepalive
+            @socket_cache.revoke
+          end
```

Comment:
Following code is needed?
```
else
  sock.close rescue nil
end
```

Reply:
Thanks. I didn't mean to delete the code. https://github.com/fluent/fluentd/pull/2393/commits/d3cb12013f13c4b849f1359d04ca8bf962b7af7b